agent/consul/config_endpoint.go

```diff
@@ -142,7 +142,7 @@ func (c *ConfigEntry) Get(args *structs.ConfigEntryQuery, reply *structs.ConfigE
 
 			reply.Index = index
 			if entry == nil {
-				return nil
+				return errNotFound
```

Comment:
This is the change that needs to be applied to all the other RPC endpoints to take advantage of this enhancement.  Currently it is only implemented on this one `ConfigEntry.Get` endpoint.